src/dotnet/commands/dotnet-build/ProjectGraphNode.cs

```diff
@@ -20,7 +21,16 @@ public ProjectGraphNode(Task<ProjectContext> projectContext, IEnumerable<Project
             IsRoot = isRoot;
         }
 
-        public ProjectContext ProjectContext { get { return _projectContextCreator.GetAwaiter().GetResult(); } }
+        public ProjectContext ProjectContext
+        {
+            get
+            {
+                using (_projectContextCreator.IsCompleted ? null : PerfTrace.Current.CaptureTiming("", "Blocking ProjectContext wait"))
```

Comment:
Most of the time task finishes before someone accesses this property, tracing is here to check for cases when it's not fast enough.